File: 1.two-sum.go
/*
 * @lc app=leetcode id=1 lang=golang
 *
 * [1] Two Sum
 */

// @lc code=start
func twoSum(nums []int, target int) []int {
	ret := []int{}
	pairTargetMap := make(map[int]int)

	for i, num := range nums {
		pairTarget := target - num
		targetIdx, ok := pairTargetMap[num]
		if ok {
			return []int{i, targetIdx}
		} else {
			pairTargetMap[pairTarget] = i
		}
	}
	return ret
}
// @lc code=end

File: 1/1.two-sum.rs
/*
 * @lc app=leetcode id=1 lang=rust
 *
 * [1] Two Sum
 */

// @lc code=start
impl Solution {
    pub fn two_sum(nums: Vec<i32>, target: i32) -> Vec<i32> {
        for i in 0..(nums.len() - 1) {
            for j in (i+1)..(nums.len()) {
                if nums[i] + nums[j] == target {
                    return vec![i.try_into().unwrap(), j.try_into().unwrap()]
                }
            }
        }
        return vec![]
    }
}
// @lc code=end

